### PR TITLE
Connect register frontend and backend.

### PR DESCRIPTION
**Pull request by shiyi001**

### CODE REVIEW

File: backend/graminsta/core/services.py [hunk 1]
```diff
@@ -27,8 +27,13 @@ def create_userinfo(validated_data):
...
     userinfo_serializer.is_valid(raise_exception=True)
     user_data = validated_data.pop('user')
-    user = UserSerializer.create(
-        UserSerializer(), validated_data=user_data)
+    user = auth_models.User.objects.create_user(
+        username=user_data["username"],
+        first_name=user_data["first_name"],
+        last_name=user_data["last_name"],
+        email=user_data["email"],
+        password=user_data["password"],
```
Comment (Time1ess) [inline backend/graminsta/core/services.py]: I think you can just use `User.objects.create_user(**user_data)`.

File: flutter/Graminsta/lib/auth/signup.dart [hunk 2]
```diff
@@ -1,12 +1,71 @@
...
+      'first_name': _firstnameController.text,
+      'last_name': _lastnameController.text,
+      'email': _emailAddressController.text,
+      'password': password1,
+    });
+    int gender = (_genderController.text == 'M') ? 0 : 1;
+    int age = int.parse(_ageController.text);
+    UserInfo userInfo = UserInfo(user: user, gender: gender, age: age);
+
+    final mess = await AuthService.signUp(userInfo);
```
Comment (Time1ess) [inline flutter/Graminsta/lib/auth/signup.dart]: nit: `mess` --> `response`?

File: flutter/Graminsta/lib/auth/signup.dart [hunk 3]
```diff
@@ -1,12 +1,71 @@
...
+    if (mess["status"] == "success") {
+      Navigator.of(context).pushNamedAndRemoveUntil(
+        '/auth',
+        (route) => route == null,
+      );
+    } else if (mess["status"] == 'fail') {
+      String errMessage = 'FAILED!\n';
+      if (mess.containsKey("user")) {
```
Comment (Time1ess) [inline flutter/Graminsta/lib/auth/signup.dart]: Use single quotation marks, ditto for others.

File: flutter/Graminsta/lib/models/user.dart [hunk 4]
```diff
@@ -4,16 +4,49 @@ class User {
...
+  final String password;
 
-  User({this.id, this.username, this.firstName, this.lastName});
+  User(
+      {this.id,
+      this.username,
+      this.firstName,
+      this.lastName,
+      this.email,
+      this.password});
```
Comment (Time1ess) [inline flutter/Graminsta/lib/models/user.dart]: Add comma before `}`

**Review by Time1ess — APPROVED** [review]

**Review by ArianaGu — APPROVED** [review]

**Pull request merged**